### ISSUE TITLE
Export Deck to Heathstone does not work after Hearthstone 4.2.0 update patch

### ISSUE THREAD

**Issue by tknights2015** [Epix37/Hearthstone-Deck-Tracker#2209]
After updating Hearthstone with the 4.2.0 patch exporting a deck to HS, results in only the name of the deck being imported. The rest is shown export incomplete. It appears to be showing the cards in the search box and the collection, but it does not add the cards to the deck.

![hdt 3_14](https://cloud.githubusercontent.com/assets/16232528/13757163/c2b03a52-e9e8-11e5-932a-e5f7ca3d5680.JPG)

Log file
[log_1432016-132843.txt](https://github.com/Epix37/Hearthstone-Deck-Tracker/files/172731/log_1432016-132843.txt)

**Comment by Roflstompkillr** [comment]
I have this same issue and after watching it work it appears when it types the card into the search it is also adding the artists name

**Comment by Sosunser** [comment]
The simplest fix seems to be adding `artist:` before each word in the artist's name.

**Comment by lexblow** [comment]
Same here. Exporting not working, need help.

**Comment by nzifnab** [comment]
Yep, it's always had artist names in the search to avoid ambiguity, but looks like Blizzard made a change to how the search accepts artist name parameters

**Comment by Epix37** [comment]
Working on it!

**Comment by Epix37** [comment]
Crap. "artist" only works with the english client.

**Comment by jeyraof** [comment]
Is this hotfix? when commit for solving this issue would release?
many thanks.

**Comment by Epix37** [comment]
Making sure I got all the "artist" translations correct now. 

(Updating HS like 12 times, fun...)

v0.13.15 should be out in a few hours

**Comment by jeyraof** [comment]
Great! Thank you @Epix37 . :100:

**Comment by Epix37** [comment]
Pre-release is out.

Download it [here](https://github.com/Epix37/Hearthstone-Deck-Tracker/releases/tag/v0.13.15) or enable `options (advanced) > tracker > settings > check for BETA updates` and restart HDT.

**Comment by dvide** [comment]
You could also use 'attack:7' to search for Stalagg now too, and 'attack:4' for Feugen.

**Comment by tknights2015** [comment]
Export in the prerelease works, but I had to export the deck deck 4 times before all the cards were exported correctly ( less than 30 cards first 3 tries) Successively more cards imported in each subsequent export from HDT to HS

**Comment by Epix37** [comment]
Do any of the [standard suggestions](https://github.com/Epix37/Hearthstone-Deck-Tracker/wiki/FAQ#exporting) help with that?

**Comment by dvide** [comment]
Pre-release seems to work fine for me on english client.

**Comment by tknights2015** [comment]
I had done this previously
Windows 7
Speed is normal
Wait 2 seconds.

The post above was not the normal 1 or 2 cards that are missing.  If I can remember correctly, the 1st import only imported no cards, the 2nd import 17,  the 3rd import 20 something and the last import all 30.

Tried another class deck. Imported 28 of 30 cards, but another issue. Imported 6 incorrect cards that were not in the deck. Deleted the deck and imported the same deck again and imported correctly

**Comment by imsomony** [comment]
@Epix37 the beta/pre-release is working fine for me, thank you so much, you're the best! I imported new decks for every hero without any trouble (minus the Feugen/Stalagg issue, no biggie). In case it helps, I'm NA region, Win10 Home.

**Comment by GiggliG** [comment]
Another confirmation here: beta is working fine for me.

**Comment by Epix37** [comment]
Good. Any other issue with the pre-release aside from cards not being added to the deck (e.g. elise)?

**Comment by raphirm** [comment]
pre release doesn't work for me unfortunately. i tried to import reno-lock and somehoe it imports all cards twice event it shouldn't happen

**Comment by Epix37** [comment]
@raphirm Please do the following:
- enable debug logging 
    - `options (advanced) > tracker  > settings > show log`
    - set `logging > log level` to `1`
- restart HDT and Hearthstone and reproduce the problem 
- post the log here ([see here](https://github.com/Epix37/Hearthstone-Deck-Tracker/wiki/Reporting-Issues))

**Comment by raphirm** [comment]
@Epix37 

thanks for the fast response. I did to as requested, here are the debug logs:
http://pastebin.com/w8XspSBq

and this is the outcome in hearthstone:
![Uploading Hearthstone Screenshot 03-15-16 14.24.17.png…]()

**Comment by Epix37** [comment]
Uh... veeery weird... it's definitely only sending one click. Not sure what the OS/Hearthstone is doing here.
You did not have this issue before the Hearthstone update?

Try lowering the `options (advanced) > tracker > exporting > speed` to `normal` or `slow`.

**Comment by Epix37** [comment]
Should be fixed with v0.13.16.

Let me know if there are remaining issues.

Did not try to fix the problem where it adds two cards instead of one as I have not yet any idea why that would happen.

**Comment by tknights2015** [comment]
Decks are importing with 13.16 however still the issue with the first import not importing many cards on the firsts import. example:

log file http://pastebin.com/F3bN5pat
Import 1
![import 1 11_09](https://cloud.githubusercontent.com/assets/16232528/13787176/7dda2fe2-eaa0-11e5-9c8c-05dd6d7ecd74.JPG)
Import 2 - no changes with HDT. deleted the first deck and imported again.
![import 11_13](https://cloud.githubusercontent.com/assets/16232528/13787196/93d9ca3c-eaa0-11e5-82d4-837903e728a6.JPG)

**Comment by Epix37** [comment]
My best guess is your hearthstone client can't keep up. Try settings the exporting speed to slow / very slow.

**Comment by tknights2015** [comment]
I slowed it down to very slow and wait 3 seconds.  Did 2 imports and got 27 cards on each. So that is close to importing all cards.

**Comment by Epix37** [comment]
Ok, I'm fairly certain then that it is due to a performance problem on your end. Not much I can do to fix that sadly.

If you want, you can try slowing it down even further by editing the `DeckExportDelay` value in the config (`options > tracker > settings > open appdata folder, config.xml`).

**Comment by tknights2015** [comment]
Thank you. I can live with it. Just did not have this problem before the HS update. Maybe missing 1-3 cards, but no so many.  In any case thanks for the tips and love the HDT.  Appreciate all that you do.

**Comment by depechenode** [comment]
today's patch I can no longer import decks either.  On first click it sets the mana gem to 0 then does not import any cards and states I am missing every card.

**Comment by depechenode** [comment]
ok my work around was setting Export speed to 3 seconds before starting.  This fixed all my issues on export

**Comment by avatarx0** [comment]
Deck exporting was working 100% on Release v0.13.16 with very slow (150ms) speed and force clear search box option checked.

Release v0.13.17 fixed deck exporting after Hearthstone was updated with more deck slots but it's not 100% perfect like 0.13.16. I have to edit some cards after the deck tracker exports cards.

**Comment by ThorsAngerVaNeT** [comment]
I have an issue with exporting on Release v0.13.17 ... HDT just paste "v" in search field for every card...

**Comment by tknights2015** [comment]
Export on version 13.7 works on exporting for me also.

**Comment by Epix37** [comment]
@ThorsAngerVaNeT see #2260 for that issue. Will be fixed with the next release.
